src/routes/relay/relay.controller.spec.ts

```diff
@@ -0,0 +1,1646 @@
+            .encode();
+          const taskId = faker.string.uuid();
+          networkService.get.mockImplementation((url) => {
+            switch (url) {
+              case `${safeConfigUrl}/api/v1/chains/${chainId}`:
+                return Promise.resolve({ data: chain, status: 200 });
+              default:
+                fail(`Unexpected URL: ${url}`);
+            }
+          });
+          networkService.post.mockImplementation((url) => {
+            switch (url) {
+              case `${relayUrl}/relays/v2/sponsored-call`:
+                return Promise.resolve({ data: { taskId }, status: 200 });
+              default:
+                fail(`Unexpected URL: ${url}`);
+            }
+          });
+
+          await request(app.getHttpServer())
+            .post(`/v1/chains/${chain.chainId}/relay`)
+            .send({
+              to,
+              data,
+            })
+            .expect(201)
+            .expect({
+              taskId,
+            });
+        });
+
+        it('should return 201  when creating an official L2 Safe', async () => {
+          const chainId = faker.helpers.arrayElement(supportedChainIds);
+          const chain = chainBuilder().with('chainId', chainId).build();
+          const version = '1.3.0';
+          const owners = [
+            getAddress(faker.finance.ethereumAddress()),
+            getAddress(faker.finance.ethereumAddress()),
+          ];
+          const singleton = getSafeL2SingletonDeployment({
+            version,
+            network: chainId,
+          })!.networkAddresses[chainId];
+          const to = faker.finance.ethereumAddress();
+          const data = createProxyWithNonceEncoder()
+            .with('singleton', getAddress(singleton))
+            .with('initializer', setupEncoder().with('owners', owners).encode())
+            .encode();
+          const taskId = faker.string.uuid();
+          networkService.get.mockImplementation((url) => {
+            switch (url) {
+              case `${safeConfigUrl}/api/v1/chains/${chainId}`:
+                return Promise.resolve({ data: chain, status: 200 });
+              default:
+                fail(`Unexpected URL: ${url}`);
+            }
+          });
+          networkService.post.mockImplementation((url) => {
+            switch (url) {
+              case `${relayUrl}/relays/v2/sponsored-call`:
+                return Promise.resolve({ data: { taskId }, status: 200 });
+              default:
+                fail(`Unexpected URL: ${url}`);
+            }
+          });
+
+          await request(app.getHttpServer())
+            .post(`/v1/chains/${chain.chainId}/relay`)
+            .send({
+              to,
+              data,
+            })
+            .expect(201)
+            .expect({
+              taskId,
+            });
+        });
+      });
+    });
+
+    describe('Transaction validation', () => {
+      // TODO: Return a 422 when the transactions are invalid
```

Comment:
I'm aware of this. I want to revisit it in a separate PR to not increase the complexity of this one.

Reply:
I've opened a [separate PR](https://github.com/safe-global/safe-client-gateway/pull/1154) implementing this.